README.md

```diff
@@ -11,99 +11,205 @@
+```
+GET http://localhost:5000/api/Actor
+```
 
-- [Code Of Conduct](./CODE_OF_CONDUCT.md)
-- [Security](./SECURITY.md)
-- [Contributing](./CONTRIBUTING.md)
-
-If you want to propose a completely new feature, please create an RFC item. Good examples of how to create RFC can be found here:
-
-- [Rust RFC Template](https://github.com/rust-lang/rfcs/blob/master/0000-template.md)
-- [Python PEP Guidance](https://www.python.org/dev/peps/pep-0001/#what-belongs-in-a-successful-pep)
-
-## References
+### Walk-through video
 
-- [Microsoft REST API Guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/Guidelines.md)
-- [Microsoft Azure REST API Guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md)
-- [GraphQL](https://graphql.org/)
+<a href="https://www.youtube.com/watch?v=xAlaoDQolLw" target="_blank">
+  <img src="https://img.youtube.com/vi/xAlaoDQolLw/0.jpg" alt="Play Video" width="280" />
+</a>
+
+Demo source code: [startrek](https://aka.ms/dab/startrek)
+
+## Overview
+
+| Category       | Features |
+|----------------|----------|
+| **Database Objects** | • NoSQL collections<br>• RDBMS tables, views, stored procedures |
+| **Data Sources** | • SQL Server & Azure SQL<br>• Azure Cosmos DB<br>• PostgreSQL<br>• MySQL |
+| **REST** | • `$select` for projection<br>• `$filter` for filtering<br>• `$orderBy` for sorting |
+| **GraphQL** | • Relationship navigation<br>• Data aggregation<br>• Multiple mutations |
+| **Telemetry** | • Structured logs<br>• OpenTelemetry<br>• Application Insights<br>• Health endpoints |
+| **Advanced** | • Pagination<br>• Level 1 (in-memory) cache<br>• Level 2 (Redis) cache |
+| **Authentication** | • OAuth2/JWT<br>• EasyAuth<br>• Entra ID |
+| **Authorization** | • Role-based support<br>• Entity permissions<br>• Database policies |
+| **Developer** | • Cross-platform CLI<br>• Swagger (REST)<br>• Banana Cake Pop (GraphQL)<br>• Open Source<br>• Hot Reload |
+
+## How does it work?
+
+This diagram shows how DAB works. DAB dynamically creates endpoints from your config file. It translates HTTP requests to SQL, returns JSON, and auto-pages results.
+
+```mermaid
+sequenceDiagram
+    actor Client
+
+    box Data API builder (DAB)
+        participant Endpoint
+        participant QueryBuilder
+    end
+
+    participant Configuration as Configuration File
+
+    box Data Source
+        participant DB
+    end
+
+    Endpoint->>Endpoint: Start
+    activate Endpoint
+        Endpoint->>Configuration: Request
+        Configuration-->>Endpoint: Configuration
+        Endpoint->>DB: Request
+        DB-->>Endpoint: Metadata
+            Note over Endpoint, DB: Some configuration is validated against the metadata
+        Endpoint-->>Endpoint: Configure
+    deactivate Endpoint
+    Client-->>Endpoint: HTTP Request
+    activate Endpoint
+        critical
+        Endpoint-->>Endpoint: Authenticate
+        Endpoint-->>Endpoint: Authorize
```

Comment:
From my observations, the authorization piece is actually executed during child resolver execution. After the SQL query is executed by root resolver, HotChocolate then checks if the user is authorized for that field. It may be a nitpick to call out that nuance though.

Reply:
We also check if the role in which the request is being executed as has permissions to access to perform the given permission on the fields specified in the query - before actually executing the query - at least with REST APIs. 

For GraphQL, looks like the middleware GraphQLAuthorizationHandler invokes `IsInHeaderDesignatedRole` . Are you sure, this middleware is invoked after the query is executed?

Reply:
Forgot about the REST piece. 
For GraphQL, yes GraphQLAuthorizationHandler.AuthorizeAsync is invoked during field resolution - the authorize directive on field is passed as input to it.